backend/organization/models/members.py

```diff
@@ -114,3 +114,76 @@ class Meta:
+                                                     ,null=False
+                                                     ,verbose_name="Membership Request Target Type",
+                                                      help_text="The type of entity a user requested to join (organization, project...)")
+
+    target_project =  models.ForeignKey(Project
+                                            ,null=True
+                                            ,on_delete=models.CASCADE
+                                            ,help_text="Points to the requested project"
+                                            ,verbose_name="Target Project of Membership Request"
+                                            ,related_name='projects')
+
+    target_organization =  models.ForeignKey(Organization
+                                            ,null=True
+                                            ,on_delete=models.CASCADE
+                                            ,help_text="Points to the requested organization"
+                                            ,related_name='organizations'
+                                            ,verbose_name='Target Ogranization of Membership request')
+
+    availability = models.ForeignKey(Availability
+                                     ,null=False
+                                     ,on_delete=models.CASCADE
+                                     ,help_text="Points to the Availability offered by the user"
+                                     ,verbose_name='requested membership availability'
+                                     ,related_name='availabilities'
+                                     )
+
+
+    requested_at = models.DateTimeField(help_text="Time of request"
+                                        ,verbose_name="Requested at"
+                                        ,default=None
+                                        ,null=False
+                                        )
+    approved_at = models.DateTimeField(help_text="Time of request approval"
+                                        ,verbose_name="Approved at"
+                                        ,default=None
+                                        ,null=True
+                                        )
+    rejected_at = models.DateTimeField(help_text="Time of request rejection"
+                                        ,verbose_name="Rejected at"
+                                        ,default=None
+                                        ,null=True
+                                        )
+
+    message = models.TextField(help_text="Message associated with the request",
+        verbose_name="Request Message",
+        max_length=4096,
+        null=True,
+        blank=True)
+
+    request_status = models.SmallIntegerField(
+        choices=[(x.value,x.name) for x in RequestStatus],
+        default=RequestStatus.PENDING.value
+
+    )
+
```

Comment:
Based on the logic it looks like a member can only request a project once. Why don't we add index here to set `user` and `project` or `user` and `organization` are unique together. I see a lot of code checking for record == 0 or record != 1. 
Resource: https://docs.djangoproject.com/en/3.2/ref/models/options/#unique-together

Reply:
You can unique constraint under `class Meta`. Look other models in the code for reference.